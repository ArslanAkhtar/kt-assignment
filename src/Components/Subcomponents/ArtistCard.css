.main-card {
  width: 300px;
  margin-top: 16px;
  cursor: pointer;
}
.main-select {
  width: 120px;
}
