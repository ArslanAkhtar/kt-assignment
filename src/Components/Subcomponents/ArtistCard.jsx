import React, { Fragment } from "react";
import { Skeleton, Card, Avatar, Divider, Row, Col, Empty } from "antd";

const ArtistCard = ({ data, loading }) => {
  if (data.err === "Service Error")
    return (
      <Row>
        <Col span={24}>
          <Empty style={{ marginTop: 24 }} />
        </Col>
      </Row>
    );
  const { Meta } = Card;
  return (
    <Fragment>
      <Row>
        <Divider />
      </Row>
      <Row>
        <Col span={24}>
          <Card style={{ width: 300, marginTop: 16 }}>
            <Skeleton loading={loading} avatar active>
              <Meta
                avatar={<Avatar src={data && data.image_url} />}
                title={data && data.name}
                description={data && data.facebook_page_url}
              />
            </Skeleton>
          </Card>
        </Col>
      </Row>
    </Fragment>
  );
};

export default ArtistCard;
