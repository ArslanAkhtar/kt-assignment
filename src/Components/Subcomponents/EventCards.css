.row {
  padding: 10px 0px;
  display: flex;
  justify-content: space-between;
}
.left {
  display: flex;
  flex-direction: column;
  width: 48%;
}
.right {
  display: flex;
  flex-direction: column;
  width: 48%;
}
span.heading {
  font-weight: bold;
}
