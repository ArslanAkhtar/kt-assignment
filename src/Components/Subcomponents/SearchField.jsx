import React, { Fragment } from "react";
import { Input, Row, Col, Typography, Space } from "antd";

const SearchField = ({ setData, setLoading }) => {
  const { Search } = Input;
  const { Title } = Typography;

  const onSearch = async (value) => {
    if (value === "") {
      setData(null);
      return;
    }
    try {
      setLoading(true);
      const response = await fetch(
        `https://rest.bandsintown.com/artists/${value}?app_id=test`
      );
      const data = await response.json();
      if (data) setLoading(false);
      setData({ ...data, err: "" });
    } catch {
      setLoading(false);
      setData({ err: "Service Error" });
    }
  };

  return (
    <Fragment>
      <Row>
        <Col span={24} offset={9}>
          <Title>Search Artist</Title>
        </Col>
      </Row>
      <Row>
        <Col span={12} offset={6}>
          <Search
            placeholder="Search Artist"
            allowClear
            enterButton="Search"
            size="large"
            onSearch={onSearch}
          />
          <Space />
        </Col>
      </Row>
    </Fragment>
  );
};

export default SearchField;
