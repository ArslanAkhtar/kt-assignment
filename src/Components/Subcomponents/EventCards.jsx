import React, { Fragment } from "react";
import { Card, Divider, Row, Col, Empty } from "antd";
import Grid from "@material-ui/core/Grid";

import "./EventCards.css";

const EventCards = ({ eventsData }) => {
  if (eventsData.length === 0) {
    return (
      <Row>
        <Col span={24}>
          <Empty style={{ marginTop: 24 }} />
        </Col>
      </Row>
    );
  }
  const date = (dateTime) => {
    const setDateTime = new Date(dateTime).toDateString();
    return setDateTime;
  };
  return (
    <Fragment>
      <Row>
        <Divider />
      </Row>

      <Grid container spacing={4}>
        {eventsData.map((card) => (
          <Grid item key={card.id} xs={12} sm={6} md={4}>
            <Card title="EVENT DETAILS" bordered={true}>
              <div className="row">
                <div className="left">
                  <span className="heading">Country</span>
                  <span className="detail">{card.venue.country}</span>
                </div>
                <div className="right">
                  <span className="heading">City</span>
                  <span className="detail">{card.venue.city}</span>
                </div>
              </div>
              <div className="row">
                <div className="left">
                  <span className="heading">Venue</span>
                  <span className="detail">{card.venue.name}</span>
                </div>
                <div className="right">
                  <span className="heading">Date</span>
                  <span className="detail">{date(card.datetime)}</span>
                </div>
              </div>
            </Card>
          </Grid>
        ))}
      </Grid>
    </Fragment>
  );
};

export default EventCards;
