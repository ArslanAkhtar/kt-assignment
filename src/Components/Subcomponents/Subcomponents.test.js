import { shallow } from "enzyme";
import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import ArtistCard from "./ArtistCard";
import EventCards from "./EventCards";
import SearchField from "./SearchField";
configure({ adapter: new Adapter() });

it("renders ArtistCard without crashing", () => {
  shallow(<ArtistCard data={{ err: "" }} loading={false} artistName={""} />);
});

it("renders ArtistCard correctly with Snapshot", () => {
  const tree = shallow(
    <ArtistCard data={{ err: "" }} loading={false} artistName={""} />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it("renders EventCards without crashing", () => {
  shallow(<EventCards eventsData={[]} />);
});

it("renders EventCards correctly with Snapshot", () => {
  const tree = shallow(<EventCards eventsData={[]} />);
  expect(toJson(tree)).toMatchSnapshot();
});

describe("Event Card Text Loaded", () => {
  it("contains Text", () => {
    const wrapper = shallow(<EventCards eventsData={[]} />);
    expect(wrapper.contains("EVENT DETAILS"));
  });
});

describe("Event Card Text Loaded Country", () => {
  it("contains Country", () => {
    const wrapper = shallow(<EventCards eventsData={[]} />);
    expect(wrapper.contains("Country"));
  });
});

describe("Event Card Text Loaded City", () => {
  it("contains City", () => {
    const wrapper = shallow(<EventCards eventsData={[]} />);
    expect(wrapper.contains("City"));
  });
});

describe("Event Card Text Loaded Venue", () => {
  it("contains Venue", () => {
    const wrapper = shallow(<EventCards eventsData={[]} />);
    expect(wrapper.contains("Venue"));
  });
});

describe("Event Card Text Loaded Date", () => {
  it("contains Date", () => {
    const wrapper = shallow(<EventCards eventsData={[]} />);
    expect(wrapper.contains("Date"));
  });
});

it("renders SearchField without crashing", () => {
  shallow(<SearchField artistName={""} />);
});

it("renders SearchField correctly with Snapshot", () => {
  const tree = shallow(<SearchField artistName={""} />);
  expect(toJson(tree)).toMatchSnapshot();
});

describe("SearchField Text Loaded", () => {
  it("contains Text", () => {
    const wrapper = shallow(<SearchField artistName={""} />);
    expect(wrapper.contains("Search Artist"));
  });
});
