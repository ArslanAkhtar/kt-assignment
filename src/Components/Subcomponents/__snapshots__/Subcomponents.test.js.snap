// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArtistCard correctly with Snapshot 1`] = `
<Fragment>
  <Row>
    <Divider />
  </Row>
  <Row>
    <Col
      offset={6}
      span={3}
    >
      <Title
        level={5}
      >
        Select Event Type:
      </Title>
    </Col>
    <Col
      span={4}
    >
      <ForwardRef(InternalSelect)
        className="main-select"
        defaultValue="all"
        onChange={[Function]}
      >
        <Option
          value="all"
        >
          All
        </Option>
        <Option
          value="upcoming"
        >
          Upcoming
        </Option>
        <Option
          value="past"
        >
          Past
        </Option>
      </ForwardRef(InternalSelect)>
    </Col>
  </Row>
  <Row>
    <Col
      span={24}
    >
      <Card
        className="main-card"
        onClick={[Function]}
      >
        <Skeleton
          active={true}
          avatar={true}
          loading={false}
          paragraph={true}
          title={true}
        >
          <Meta
            avatar={
              <Avatar
                shape="circle"
                size="default"
              />
            }
          />
        </Skeleton>
      </Card>
    </Col>
  </Row>
</Fragment>
`;

exports[`renders EventCards correctly with Snapshot 1`] = `
<Row>
  <Col
    span={24}
  >
    <Empty
      style={
        Object {
          "marginTop": 24,
        }
      }
    />
  </Col>
</Row>
`;

exports[`renders SearchField correctly with Snapshot 1`] = `
<Fragment>
  <Row>
    <Col
      offset={9}
      span={24}
    >
      <Title>
        Search Artist
      </Title>
    </Col>
  </Row>
  <Row>
    <Col
      offset={6}
      span={12}
    >
      <Search
        allowClear={true}
        enterButton="Search"
        onSearch={[Function]}
        placeholder="Search Artist"
        size="large"
      />
      <Space />
    </Col>
  </Row>
</Fragment>
`;
